rmf_demos_fleet_adapter/rmf_demos_fleet_adapter/fleet_manager.py

```diff
@@ -475,6 +533,16 @@ def get_robot_state(self, robot: State, robot_name):
             data['replan'] = True
         else:
             data['replan'] = False
+        if (robot.state.mode.mode == RobotMode.MODE_ACTION_COMPLETED):
+            self.get_logger().info(
+                f'Robot [{robot_name} completed performing its action')
+            completed_cmd_id = 0
+            msg = self._make_mode_request(robot_name, completed_cmd_id,
+                                          RobotMode.MODE_IDLE)
+            # Mark action execution as finished
+            self.action_completed_pub.publish(msg)
```

Comment:
While digging at the code I noticed we would publish an message to denote action completion here for every type of action. This is fine but I noticed that in the subscription in the fleet adapter side we still have `teleop` specific code, and when a message is received here we trigger a toggle_teleop call https://github.com/open-rmf/rmf_demos/pull/225/files#diff-1fe77ef1a1d2d883027183fc27ba21f5462e2cb461be28f986296a28eebf6085R283-R285, it seems like this is not the correct behavior and we should only toggle teleop when the action that was being performed is indeed teleop?

Reply:
Yup you're right, added a check for an existing `Teleoperation` object stored before calling the `toggle_teleop` endpoint in [04a6fb9](https://github.com/open-rmf/rmf_demos/pull/225/commits/04a6fb939e3e206e67d56f5d31e444f3d96ed7b1). A robot can only perform one action at a time, so if `self.teleoperation` is not None, it means that the current ongoing action is a teleop action. Else, it'd be some other perform action.